src/storage/src/decode/mod.rs

```diff
@@ -100,8 +100,8 @@ pub fn render_decode_cdcv2<G: Scope<Timestamp = Timestamp>>(
                     None => continue,
                 };
                 let (mut data, schema, _) = match resolver.resolve(&*value).await {
-                    Ok(ok) => ok,
-                    Err(e) => {
+                    Ok(Ok(ok)) => ok,
+                    Ok(Err(e)) | Err(e) => {
```

Comment:
this is cdcv2, so not as important, but this second case should restart the dataflow right? probably worth a TODO